enderio-base/src/main/java/com/enderio/api/energy/IMachineEnergy.java

```diff
@@ -0,0 +1,67 @@
+package com.enderio.api.energy;
+
+import net.minecraft.core.Direction;
+import org.jetbrains.annotations.Nullable;
+
+public interface IMachineEnergy {
```

Comment:
Looking for some feedback specifically to do with this interface. I added it as a way to separate the energy transfer and energy storage concepts apart, instead of having a dedicated class for storing energy, however I feel maybe moving back to energy storage being its own class, similarly to how items and fluids are handled might just make more sense from a design perspective. If this were to happen, how would we deal with IMachineEnergy. I added it as it has some clear additions over forge's own IEnergyHandler, namely consumption rates, leak rates etc. which are important to enderio storages. Do we just get rid of this interface and merge all this functionality together in a single class?

My motivation for bringing the energy storage *into* the block entity was to remove the need for so many overrides, we can now directly influence the insert/extract behaviour in a BE for example, without having to override the EnergyStorage class or having awkward overloads.

Looking for feedback basically as to whether this is an overcomplicated change or if its better to have separation between "Forge Energy" and "EnderIO Energy Storage"

Reply:
I'm gonna try a different approach, I think it'll be easier to understand and have a cleaner impl.